hydra-task/src/main/java/com/addthis/hydra/task/map/MapFeeder.java

```diff
@@ -103,20 +103,23 @@ public MapFeeder(StreamMapper task, TaskDataSource source, int feeders) {
         }
 
         try {
+            boolean interrupt = false;
             if (source.isEnabled()) {
                 while (fillBuffer()) {
                     if (Thread.interrupted()) {
+                        interrupt = true;
                         closeSourceIfNeeded();
                     }
                 }
             }
+            final boolean interrupted = interrupt;
```

Comment:
That's fine I would consider that a benign race condition. There is no harm done if the onTaskComplete fires. It just means we could open a source then is immediately closed.